### PR TITLE
fix signed/unsigned printf format mismatches

### PR DESCRIPTION
**Pull request by kmilos**
Found w/ cppcheck 2.10

### CODE REVIEW

File: src/iop/lut3d.c [hunk 1]
```diff
@@ -842,7 +842,7 @@ uint16_t calculate_clut_cube(const char *const filepath, float **clut)
   }
   if(out_of_range_nb)
   {
-    fprintf(stderr, "[lut3d] warning - %d values out of range [0,1]\n", out_of_range_nb);
+    fprintf(stderr, "[lut3d] warning - %u values out of range [0,1]\n", out_of_range_nb);
```
Comment (ralfbrown) [inline src/iop/lut3d.c:845]: Since we're changing the code anyway, should this be switched to dt_print?

File: src/iop/segmentation.h [hunk 2]
```diff
@@ -88,7 +88,7 @@ static inline int _get_segment_id(dt_iop_segmentation_t *seg, const size_t loc)
 {
   if(loc >= (size_t)(seg->width * seg->height))
   {
-    fprintf(stderr, "[_get_segment_id] out of range access loc=%lu in %ix%i\n", loc, seg->width, seg->height);
+    fprintf(stderr, "[_get_segment_id] out of range access loc=%zu in %ix%i\n", loc, seg->width, seg->height);
```
Comment (ralfbrown) [inline src/iop/segmentation.h:91]: Use dt_print?

File: src/iop/lut3d.c (hunk 1 above)
Comment (TurboGit) [reply src/iop/lut3d.c:845]: A good idea, `dt_print(DT_DEBUG_ALWAYS,  "[lut3d]...`

Comment (kmilos) [reply src/iop/lut3d.c]: Done for all the output for the two modules in question.

**Comment by kmilos** [comment]
Something's up w/ the GH runners, so will have have to be re-run once they're back...

**Comment by TurboGit** [comment]
@kmilos : CI restarted...

**Review by TurboGit — APPROVED** [review]
Looks good, thanks!

**Pull request merged**